app/src/org/commcare/UpdateToPrompt.java

```diff
@@ -0,0 +1,158 @@
+package org.commcare;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.util.Base64;
+
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.Externalizable;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * Created by amstone326 on 4/13/17.
+ */
+
+public class UpdateToPrompt implements Externalizable {
```

Comment:
I actually didn't consider it at all. I think you're right that it would be marginally cleaner, but I think it probably doesn't have enough of a clear advantage to be worth the refactor right now. Let me know if you disagree, though.